### ISSUE TITLE
Resolution progress indication.

### ISSUE THREAD

**Issue by alloy** [rubygems/rubygems#705]
I’ve tried to search the existing tickets and the [support site](http://help.rubygems.org) and except for a suggestion to add a progress bar, I haven’t seen my specific issue addressed. I hope this is the right place for a discussion ticket, otherwise please tell me where to move it.

---

We work on a dependency manager for Objective-C projects, called [CocoaPods](https://github.com/CocoaPods/CocoaPods), and as such we have a lot of users who have never used RubyGems before. A question that’s coming up more and more frequent is regarding indication that RubyGems is doing anything, or rather the lack thereof.

Inevitably this leads to people cancelling the installation a couple of times before they realise that they need to let it run or turn to twitter to complain about our installation process. So while I prefer less output over more, the user-experience could do with a bit of improving, even if it’s not really needed for the experienced RubyGems user.

This is _not_ about progress indication once gems are being downloaded, but about the period _between_ entering the install command and the first sign of output. (An educated guess is that during this period it’s downloading the marshalled specs and resolving versions etc.)

We could suggest users to use the verbose option, but that’s not pretty and/or user-friendly. My current suggestion would be to print some helpful message _immediately_ after entering the command, or, in case installing gems can sometimes be finished in mere seconds, to add a short timeout before printing, i.e. show helpful message after 3-5 seconds in case gems are not being downloaded yet.

---

The output of the various phases would then look like this:

```
$ gem install xcodeproj
Updating available gems data...
```

```
Building native extensions.  This could take a while...
```

```
Successfully installed xcodeproj-0.14.1
1 gem installed
```

It doesn’t need to be a message btw, I could also imagine a dot progress reporter for each small step during that period. E.g. for each redirect during specs fetching.

---

PS: In case this is something we can agree on I’m more than willing to whip up a patch!

**Comment by copiousfreetime** [comment]
Looks like what this is asking for now is some feedback to the user during dependency resolution. Something diplayed to our users when we are interrogating the rubygems dependency api and then during dependency resolution.

**Comment by segiddins** [comment]
Molinillo has a UI hook to update every 100 ticks or so, but its not dependency resolution thats the slow part any more

**Comment by copiousfreetime** [comment]
Whatever the slow part is, I can definitely see a use for something to display for the user between the invocation of the install command and the first feedback to the user in when in non-verbose mode.

For instance on my laptop:
```
% gem install rails 
# 30 seconds of dead time before
Fetching: rack-1.6.4.gem (100%)
```
In `-V` mode
```
% gem install -V rails
HEAD https://api.rubygems.org/api/v1/dependencies
200 OK
GET https://api.rubygems.org/api/v1/dependencies?gems=rails
200 OK
# 15 seconds of dead time here
Getting SRV record failed: DNS result has no information for _rubygems._tcp.api.rubygems.org
...
```

In any case, I can see the need for some visual indicator that something is being done between the execution of the install command and the first display of `Fetching...`

**Comment by deivid-rodriguez** [comment]
I believe the last issue mentioned here might've been fixed by the same change that closed https://github.com/rubygems/rubygems/issues/2017 (shipped with rubygems 3.0). So I don't think we have too long gaps without any output anymore.-

So I think this can be closed now?

**Comment by deivid-rodriguez** [comment]
Assuming "yes" as an answer :)
